src/encoded/types/cart.py

```diff
@@ -96,3 +108,47 @@ def get_or_create_cart_by_user(context, request):
         '@type': ['result'],
         '@graph': carts or [cart]
     }
+
+
+@view_config(context=Cart.Collection, request_method='PUT', permission='save-carts', name='put-cart')
+def create_cart_by_user(context, request):
+    userid = [
+        p.replace('userid.', '')
+        for p in request.effective_principals
+        if p.startswith('userid.')
+    ]
+    if not userid:
+        raise HTTPBadRequest()
+    else:
+        userid = userid[0]
+    user = request.registry[COLLECTIONS]['user'].get(userid)
+    if not user:
+        raise HTTPBadRequest()
+    cart_name = None
+    user_carts = get_cart_objects_by_user(request, userid)
+    visible_user_carts = filter_carts_to_visible(request, user_carts)
+    countable_carts = [
+        cart
+        for cart in visible_user_carts
+        if cart['status'] != 'disabled'
+    ]
+    cart_atids = [cart['@id'] for cart in visible_user_carts] if visible_user_carts else []
```

Comment:
I don't think `if visible_user_carts else []` condition is necessary here because visible_user_carts will just be an empty cart if it doesn't exist. So cart_atids will be an empty list if you iterate over an empty list.